### ISSUE TITLE
API Help Please.  Easy question?  directory structure for API & install

### ISSUE THREAD

**Issue by glad2badad** [benkeen/generatedata#417]
Hi.  Trying to get API $POST calls to work but I think something isn't correct with my setup.

I've edited my settings.php file with $apiEnabled = true;
Other $POST calls to the httpd web server return results correctly.

My directory level is .../httpd/htdocs/api/v1/data (I manually created v1 and data.
In the api directory there are three files: index.html, process.php, and README.md

I've tried making calls, via the browser, to test:
localhost/api/v1/data ... returns "unable to connect"
localhost/api/v1 ... returns "unable to connect"

I've tried various combinations using private and public ip values in place of localhost, etc...

In the generatedata documentation there are two API reference docs.  One says to use api/v1, while the other says use api/v1/data.
https://github.com/benkeen/generatedata/tree/master/api
http://benkeen.github.io/generatedata/api.html

Which one is right?  What am I missing?

I really, REALLY, need some help with this.  It's driving me nuts.  :-/

Thanks.

**Comment by benkeen** [comment]
Hey @glad2badad - I'm so sorry I haven't been around much, it's been an absolutely crazy week and I'm spread way too thin. My apologies!

The directory structure for the API is actually "fake" - i.e. you don't need to create the folders within `/api`. The `.htaccess` folder in the repository root contains a rewrite rule that redirects all requests that contain a URL within the /api folder to the `/api/process.php` file. That file then examines what the URL is and handles the request appropriately.

I'm wondering if possibly your apache server doesn't have `mod_rewrite` installed...? It's pretty standard, but still optional so if it's not, the API routing won't work. Maybe give some of the answers to this post a go to find out:
http://stackoverflow.com/questions/7337724/how-to-check-whether-mod-rewrite-is-enable-on-server

Again, sorry for not responding sooner - I do feel bad. Let's try and sort this out.

**Comment by glad2badad** [comment]
Still working it this morning ... I've verified I do have mod_rewrite loaded (httpd).  I do NOT have a .htaccess file for my httpd web server.  Looking to create that now - not 100% certain where I need to put it (directory).  Thanks again for the help (hope you got my email response over the weekend).

**Comment by glad2badad** [comment]
Ok.  Lots of edits to my installs, configs, etc...  bottom line: I'm able to get the API response as expected. 

I ended up turning off dynamic extensions in PHP I had enabled (was getting conflicts against configure with options I had made with PHP initially.  Also added a htaccess file to the root of my web folder.  

So ... now I have one more big question.  When I use the API correctly, what can I expect in return?

Say I make a successful call to the API requesting a sample data set with basic csv export type ... won't I get a file generated?  Where will that file end up in my web directory structure?  Do I need to specify a file name somehow.

Ideally, I want this to work where I'm passing properly constructed json files to request sample data sets and returning a file that can be picked up by another process (either in a return type call or having the generated files directed to a landing zone (dir)).

Additional question for those that are using the API successfully ... what have you found to be the best method for building the curl options?  For example, I'm experimenting with several methods to pass the json post request ... (curl_file_create, file_get_contents, http_build_query).  Any of those work better than others?  Is there something easier I'm missing I should try?

Thanks again for your consideration of my questions.  I should be close to done with the pestering.  :-)
